Userland/Services/FileSystemAccessServer/ClientConnection.cpp

```diff
@@ -90,23 +90,35 @@ Messages::FileSystemAccessServer::RequestFileResponse ClientConnection::request_
     return { -1, Optional<IPC::File> {} };
 }
 
-void ClientConnection::prompt_open_file(String const& path_to_view, Core::OpenMode const& requested_access)
+void ClientConnection::prompt_open_file(i32 window_server_client_id, i32 parent_window_id, String const& path_to_view, Core::OpenMode const& requested_access)
 {
     auto relevant_permissions = requested_access & (Core::OpenMode::ReadOnly | Core::OpenMode::WriteOnly);
     VERIFY(relevant_permissions != Core::OpenMode::NotOpen);
 
     auto main_window = GUI::Window::construct();
+    main_window->set_opacity(0);
+    main_window->set_frameless(true);
+    main_window->show();
+
+    GUI::Application::the()->set_parent(window_server_client_id, parent_window_id, main_window->window_id());
```

Comment:
I'd prefer to have a method like `GUI::Application::create_dummy_window(window_server_client_id, parent_window_id)` rather than this.